vernac/auto_ind_decl.ml

```diff
@@ -435,16 +442,18 @@ let do_replace_bl mode bl_scheme_key (ind,u as indu) aavoid narg lft rgt =
                    (str "Var " ++ Id.print s ++ str " seems unknown.")
       )
     in mkVar (find 1)
-  with e when CErrors.noncritical e ->
+(* This code has no reason to work in general for an arbitrary constant *)
+(*
```

Comment:
Are there any cases that used to be correctly supported with this code that are no longer supported?

Reply:
I just tried to find one, but without success. For instance, it would have been to support an example like:
```coq
Parameter A:Type.
Parameter eq_A:A->A->bool.
Inductive I := C : A -> I.
Scheme Equality for I.
```
but it fails (with an incorrect message) due to other limitations of the code.

I tried the following which had code partially written to support it:
```coq
Section A.
Variable A:Type.
Variable eq_A:A->A->bool.
Inductive I := C : A -> I.
Scheme Equality for I.
End A.
```
But it fails also (with an anomaly).

At the end, I just preferred to add a commit to deactivate the partial support for variables and parameters.

Reply:
Congratulations in finding what is the exact way to make this working as little changes (different types, A_lb missing, A_bl missing) provokes an anomaly.

Would the following not be good enough for you:
```coq
Inductive I (A:Type) := C : A -> I A.
Scheme Equality for I.
```

Reply:
I think I originally found it by looking at the proof term that `Scheme Equality` generates, and the types of the lemmas that it generates, and mimicking that.
I'd adjust to this not working if I have to, but sometimes it's nice to be able to collect the arguments together in a section where I'm doing many things.  Also, with the new changes to type inference of inductive parameters (#7254), where the suggested fix is to make the parameters context variables, it seems kind-of annoying to have to open the section to define the inductive, close the section to define the equality scheme, and then open a new section if I want a bunch of lemmas to share the same parameters.

Reply:
Also, I can't seem to get the anomalies you're talking about with 8.8.  I just get "Error: Unexpected error during scheme creation: Attempt to save an incomplete proof" when I change the types so that things do not work.  (And there are a bunch of small changes that work, e.g., adding extra hypotheses of type `True`, using `true = eq_A x y` rather than `eq_A x y = true`, etc.  However, I do get an anomaly if I change `forall x y, x = y -> eq_A x y = true` to end with `eq_A x x = true`.)